### PR TITLE
Add support to create a custom route in a subnet for TGW

### PR DESCRIPTION
**Pull request by findmyname666**
**Reasoning**

TLDR The company where I work right now have 2 TGWs in the VPC. Each TGW is used for the different traffic.
It means I have to add a route for both TGWs to some specific subnets. Unfortunately I didn't find any way how to achieve it in the current state of the module. I noticed that one can use variables **transit_gateway_id** and **transit_gateway_routes** but this works only for one TGW. Additionally It is worth to mention that creation of VPC and TGWs is out of my control.

**Implementation notes**

Adding a new variable **custom_route_to_tgw** which is map of objects.
The map key is used to identify required subnets.
Each object contains route configuration.
Right now I'm creating a new route in the private and public subnets.
Not sure if this can be useful also for the transit_gateway and / or core_network subnets.
It is fair to mention that there is one limitation of this variable - one cannot create more routes per a subnet.
Not sure how to achieve it without too much changes.

### CODE REVIEW

**Comment by drewmullen** [comment]
LGTM. also please release as v4.3.1 @tlindsay42
